engine/crates/engine-v2/Cargo.toml

```diff
@@ -14,20 +14,28 @@ keywords = ["graphql", "engine", "grafbase"]
 workspace = true
 
 [dependencies]
+async-trait.workspace = true
 async-runtime = { workspace = true }
+base64.workspace = true
 derive_more = "0.99"
 im = "15"
 indexmap.workspace = true
+jsonwebtoken.workspace = true
```

Comment:
not sure this would work that well. We will for sure have logic that depends on fields like Apollo's `@authenticated` or `@requiresScope`. Or even just our own auth we have in v1 for read/write operations etc.. All of that is tied to the GraphQL layer

Reply:
None of that sounds like it needs to know anything about OAuth/OpenID/JWTs: `@authenticated` just needs a boolean, `@requiresScope` just needs a list of scopes, read/write operations probably also just needs some kind of parameter.

But anyway, unimportant for now.